packages/twenty-front/src/modules/auth/hooks/useAuth.ts

```diff
@@ -267,7 +281,23 @@ export const useAuth = () => {
   ]);
 
   const handleVerify = useCallback(
-    async (loginToken: string) => {
+    async (loginToken: string, email: string) => {
+      if (isEmailVerificationRequired) {
+        const maybeUser = await checkUserExistsQuery({
```

Comment:
this endpoint is protected by captcha guard

Reply:
Is that a problem? My understanding based on this snippet is that:

https://github.com/twentyhq/twenty/blob/b22a598d7d1901b940ad91be47358f696d31abf9/packages/twenty-front/src/modules/app/effect-components/PageChangeEffect.tsx#L155C1-L164C77

since the handleVerify only ever gets called from the `AppPath.SignInUp` page, the requests made will have a valid captcha token. Is that correct?

I was actually thinking of replacing the `checkUserExistsQuery` with something like `checkUserEmailVerified`, but decided to return the `emailVerified` attribute in `checkUserExistsQuery` itself to reuse what was already existing. Do you think its okay to introduce `checkUserExistsQuery`?

Reply:
Yes captcha guard is optional. People who self-host / the vast majority of instances don't use it. But for our cloud version app.twenty.com which is exposed we need it. We use Google recatpcha on our cloud. You can signup for free to try it
The excludedOperations are for the Auth Bearer verification but shouldn't affect captcha?

Reply:
Yes, that is what I realized. I tried using Cloudflare Turnstile but was having some issues, I'll see if Google recaptcha works for me.

> The excludedOperations are for the Auth Bearer verification but shouldn't affect captcha?
Exactly, I came to the same conclusion.